proposals/004-prototype-kit-extension-system.md

```diff
@@ -0,0 +1,84 @@
+# GOVUK Prototype Extensions
+
+**Closing date:** N/A
+
+**Status:** proposed
+
+## Summary
+
+The extension we’re proposing aims to make it easier for:
+
+prototype users to install and use patterns and components from any government department
+ - developers in government departments to make their custom components available for use in prototypes within their department
+ - developers in government departments to share their custom components for use in other government departments
+ - developers in government departments to extend the GOV.UK Prototype Kit
+ - developers in government departments to contribute their components to GOVUK Frontend
+
+## Problem
+
+Designers who want to use department-specific components in their prototypes have to follow complicated steps which can be intimidating and difficult to follow. In [HMCTS’s documentation](https://github.com/hmcts/frontend/blob/7b52ef51aad9b2b5453db23e3f20aeb23b73a08e/docs/installation/installing-with-npm.md) there are a number of complicated installation steps to follow which can cause errors. We want to avoid this and think that each government department will need their own version of these steps.
+
+## Proposal
+
+Developers creating components for government departments could provide an npm package containing components and a configuration file specifying the integration points required to make those components available to prototype users.
+
+An updated version of the GOV.UK Prototype Kit could support the installation of these npm packages so that designers do not have to make manual modifications of core prototype kit files.
+
+By including a [configuration file](https://github.com/hmcts/frontend/issues/31) HMCTS and HMRC components can be installed via npm and then used immediately without prototype users needing to make changes to source files.
+
+Govuk-frontend could be integrated in the same way, this both proves that the extension system is powerful enough to support govuk’s needs and removes hard-coded references to the govuk-frontend module.
```

Comment:
I really like this part of the proposal, as if we decide to do this it'll force us to test and maintain this extension system properly.

Reply:
Excellent - I totally agree.

If we can assume that `govuk-frontend` will contain a config file then it simplifies the PR I'm ready to raise to `govuk-prototype-kit`.  Simplification of the code and "dogfooding" is good IMHO.